llvm/test/CodeGen/AMDGPU/llc-pipeline.ll

```diff
@@ -523,6 +525,8 @@
 ; GCN-O1-OPTS-NEXT:      Scalarize Masked Memory Intrinsics
 ; GCN-O1-OPTS-NEXT:      Expand reduction intrinsics
 ; GCN-O1-OPTS-NEXT:      Early CSE
+; GCN-O1-OPTS-NEXT:    AMDGPU Preload Kernel Arguments
```

Comment:
Quick side question: does this pass need to run before the main CodeGenPrepare?

If not ... I have a plan to move AMDGPULowerBufferFatPointers after LoadStoreVectorizer, and, given the desire for buffer pointer alignment reform, might want to move kernel argument lowering after that so we can see function arguments as themselves.

Not exactly relevant or blocking, just thought I'd check on this

Reply:
I'd expect this to benefit from maximally canonical IR and run a bit earlier

Reply:
The pass mostly wants to run before function attributes are finalized that may influence the number of available user SGPRs. It also needs to be before LowerKernelArguments. Beyond that there may be some small benefit to lowering the hidden arguments earlier, but only if it doesn't run into the first issue.